packages/github/src/github.ts

```diff
@@ -1,158 +1,36 @@
-import {
-  graphql as GraphQL,
-  RequestParameters as GraphQLRequestParameters
-} from '@octokit/graphql/dist-types/types'
-
-import {Octokit} from '@octokit/rest'
 import * as Context from './context'
-import * as http from 'http'
-import * as httpClient from '@actions/http-client'
+import * as Utils from './internal/utils'
 
-// We need this in order to extend Octokit
-Octokit.prototype = new Octokit()
+// octokit + plugins
+import {Octokit} from '@octokit/core'
+import {OctokitOptions} from '@octokit/core/dist-types/types'
+import {restEndpointMethods} from '@octokit/plugin-rest-endpoint-methods'
+import {paginateRest} from '@octokit/plugin-paginate-rest'
 
 export const context = new Context.Context()
 
-export class GitHub extends Octokit {
-  graphql: GraphQL
-
-  /* eslint-disable no-dupe-class-members */
-  // Disable no-dupe-class-members due to false positive for method overload
-  // https://github.com/typescript-eslint/typescript-eslint/issues/291
-
-  /**
-   * Sets up the REST client and GraphQL client with auth and proxy support.
-   * The parameter `token` or `opts.auth` must be supplied. The GraphQL client
-   * authorization is not setup when `opts.auth` is a function or object.
-   *
-   * @param token  Auth token
-   * @param opts   Octokit options
-   */
-  constructor(token: string, opts?: Omit<Octokit.Options, 'auth'>)
-  constructor(opts: Octokit.Options)
-  constructor(token: string | Octokit.Options, opts?: Octokit.Options) {
-    super(GitHub.getOctokitOptions(GitHub.disambiguate(token, opts)))
-
-    this.graphql = GitHub.getGraphQL(GitHub.disambiguate(token, opts))
-  }
-
-  /**
-   * Disambiguates the constructor overload parameters
-   */
-  private static disambiguate(
-    token: string | Octokit.Options,
-    opts?: Octokit.Options
-  ): [string, Octokit.Options] {
-    return [
-      typeof token === 'string' ? token : '',
-      typeof token === 'object' ? token : opts || {}
-    ]
-  }
-
-  private static getOctokitOptions(
-    args: [string, Octokit.Options]
-  ): Octokit.Options {
-    const token = args[0]
-    const options = {...args[1]} // Shallow clone - don't mutate the object provided by the caller
-
-    // Base URL - GHES or Dotcom
-    options.baseUrl = options.baseUrl || this.getApiBaseUrl()
-
-    // Auth
-    const auth = GitHub.getAuthString(token, options)
-    if (auth) {
-      options.auth = auth
-    }
-
-    // Proxy
-    const agent = GitHub.getProxyAgent(options.baseUrl, options)
-    if (agent) {
-      // Shallow clone - don't mutate the object provided by the caller
-      options.request = options.request ? {...options.request} : {}
-
-      // Set the agent
-      options.request.agent = agent
-    }
-
-    return options
+const baseUrl = Utils.getApiBaseUrl()
+const defaults = {
+  baseUrl,
+  request: {
+    agent: Utils.getProxyAgent(baseUrl)
   }
+}
 
-  private static getGraphQL(args: [string, Octokit.Options]): GraphQL {
-    const defaults: GraphQLRequestParameters = {}
```

Comment:
Octokit now comes with an attached Graphql instance, so this is no longer needed and has been removed.